homeassistant/components/incomfort/config_flow.py

```diff
@@ -0,0 +1,51 @@
+"""Config flow support for Intergas InComfort integration."""
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import DOMAIN
+from .models import async_connect_gateway
+
+TITLE = "Intergas InComfort/Intouch Lan2RF gateway"
+
+
+class InComfortConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow to set up an Intergas InComfort boyler and thermostats."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        data_schema = {
+            vol.Required(CONF_HOST): TextSelector(
+                TextSelectorConfig(type=TextSelectorType.TEXT)
+            ),
+            vol.Optional(CONF_USERNAME): TextSelector(
+                TextSelectorConfig(type=TextSelectorType.TEXT, autocomplete="admin")
+            ),
+            vol.Optional(CONF_PASSWORD): TextSelector(
+                TextSelectorConfig(type=TextSelectorType.PASSWORD)
+            ),
+        }
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            if await async_connect_gateway(self.hass, user_input, errors):
```

Comment:
I think we should split this. This function modifies the errors dict (which is a sideeffect). Also, at the other place where it's used (init.py), we also do more calls to the device, which currently aren't covered by the try except.

Reply:
`async_connect_gateway` connects and validates there is a valid config before adding the entry. The modification of `errors` is intended.
At `__init__.py`. Errors is used to raise, and then also the additional data is fetched (`await client.update`).
I have not clear why that should be changes.

Reply:
Try blocks were split